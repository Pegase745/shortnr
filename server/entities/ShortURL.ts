import {
  Column,
  CreateDateColumn,
  Entity,
  Index,
  PrimaryColumn,
} from 'typeorm';

@Entity()
export default class ShortURL {
  @Index()
  @PrimaryColumn()
  public id: string;

  @Column()
  public url: string;

  @CreateDateColumn()
  public createdAt: Date;
}
