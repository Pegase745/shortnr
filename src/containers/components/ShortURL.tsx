import * as copyToClipboard from 'copy-to-clipboard';
import * as React from 'react';
import { Grid, Input, Segment } from 'semantic-ui-react';

interface IShortURLProps {
  shortURL: string;
  hostName: string;
}

const ShortURL: React.FC<IShortURLProps> = ({ shortURL, hostName }) => {
  const [copied, setCopied] = React.useState(false);
  const fullURL = `${hostName}r/${shortURL}`;
  const handleClick = () => {
    copyToClipboard(fullURL);
    setCopied(true);
    setTimeout(() => setCopied(false), 3000);
  };

  const actionProps = {
    color: 'teal',
    icon: 'copy',
    onClick: handleClick,
  };

  return (
    <Segment basic={true} textAlign="center">
      <Grid textAlign="center">
        <Grid.Column width={9}>
          <Input
            fluid={true}
            readOnly={true}
            action={actionProps}
            defaultValue={fullURL}
          />
        </Grid.Column>
      </Grid>
    </Segment>
  );
};

export default React.memo<IShortURLProps>(ShortURL);
