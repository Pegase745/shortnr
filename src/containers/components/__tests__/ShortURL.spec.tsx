import { shallow } from 'enzyme';
import * as React from 'react';

import ShortURL from '../ShortURL';

const defaultProps = {
  fullURL: 'https://shortnr.local/r/shortURL',
};

describe('ShortURL.tsx', () => {
  it('should render', () => {
    const app = shallow(<ShortURL {...defaultProps} />);

    expect(app).toMatchSnapshot();
  });
});
