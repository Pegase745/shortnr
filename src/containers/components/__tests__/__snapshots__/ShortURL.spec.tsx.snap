// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortURL.tsx should render 1`] = `
<Segment
  basic={true}
  textAlign="center"
>
  <Grid
    textAlign="center"
  >
    <GridColumn
      width={9}
    >
      <Input
        action={
          Object {
            "color": "teal",
            "icon": "copy",
            "onClick": [Function],
          }
        }
        defaultValue="https://shortnr.localr/shortURL"
        fluid={true}
        readOnly={true}
        type="text"
      />
    </GridColumn>
  </Grid>
</Segment>
`;
